src/components/formsBlank/forms.tsx

```diff
@@ -1,98 +1,116 @@
+      birth: birth,
+      fruit: fruit,
 
-    this.props.onClick(objData);
+      male: male,
+      agreement: agreement,
+    };
+    alert('Вы в списке');
+    formData.onSubmitPage(cardData);
+    reset();
   };
-  componentDidUpdate(): void {
-    if (this.props.clear) {
-      this.formRef.current?.reset();
-    }
-  }
-  render() {
-    return (
-      <div className={classes.formsContainer}>
-        <form ref={this.formRef} className={classes.forms} onSubmit={this.submitForm}>
-          Любитель фруктов:
-          <div className={classes.name}>
-            Введите имя:
-            <input ref={this.nameRef} type="text" name={pageData.formsField[0]} />
-            {this.props.errors.name && (
-              <p className={classes.active}>Введите корректное имя, минимум три буквы</p>
-            )}
-          </div>
-          <div>
-            Добавьте фото:
-            <input ref={this.fotoRef} type="file" name={pageData.formsField[1]} />
-            {this.props.errors.foto && <p className={classes.active}>Добавьте фото</p>}
-          </div>
-          <div>
-            Введите дату рождения:
-            <input ref={this.birthRef} type="date" name={pageData.formsField[2]} />
-            {this.props.errors.birth && <p className={classes.active}>Вы еще не родились</p>}
-          </div>
-          <div>
+
+  return (
+    <div className={classes.formsContainer}>
+      <form className={classes.forms} onSubmit={handleSubmit(onSubmit)}>
+        Любитель фруктов:
+        <label className={classes.name}>
+          Введите имя:
+          <input
+            type="text"
+            {...register('name', {
+              required: 'Введите имя',
+              minLength: {
+                value: 3,
+                message: 'Mинимум три буквы',
+              },
+            })}
+          />
+        </label>
+        <div className={classes.active}>
+          {errors?.name && (
+            <p>
+              {errors?.name?.message?.toString() || 'Введите корректное имя, минимум три буквы'}
+            </p>
+          )}
+        </div>
+        <label>
+          Добавьте фото:
+          <input
+            type="file"
+            {...register('foto', {
+              required: 'Добавьте фото',
+            })}
+          />
+        </label>
+        <div className={classes.active}>
+          {errors?.foto && <p>{errors?.foto?.message?.toString() || 'Добавьте фото'}</p>}
+        </div>
+        <label>
+          Введите дату рождения:
+          <input
+            type="date"
+            {...register('birth', {
+              required: 'Введите дату рождения',
+              validate: (value) =>
+                new Date(value).getTime() < new Date().getTime() || 'Вы еще не родились',
+            })}
+          />
+        </label>
+        <div className={classes.active}>
+          {errors?.birth && <p>{errors?.birth?.message?.toString() || 'Введите дату рождения'}</p>}
```

Comment:
just create some kind of enum for all this messages better outside the component or in some separate file 'constants'